chat-plugins/uno.js

```diff
@@ -0,0 +1,563 @@
+"use strict";
+
+const maxTime = 20; // seconds
+
+function cardHTML(card, fullsize) {
+	let surface = card.value.replace(/[^A-Z0-9\+]/g, "");
+	return `<button style="height: 140px; padding-bottom: 120px; text-align: left; font-weight: bold; border: 2px solid rgba(33 , 68 , 72 , 0.59); min-width: ${fullsize ? "80" : "30"}px; border-radius: 10px 2px 2px 3px; color: ${card.colour === "Black" ? "yellow" : "black"}; background-color: ${card.colour}" name=send value="/uno play ${card.name}">${surface}</button>`;
+}
+
+function createDeck() {
+	let colours = ["Red", "Blue", "Green", "Yellow"];
+	let values = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "Reverse", "Skip", "+2"];
+
+	let basic = [];
+
+	for (let i = 0; i < 4; i++) {
+		let colour = colours[i];
+		basic.push(...values.map(v => {
+			return {value: v, colour: colour, name: colour + " " + v};
+		}));
+	}
+
+	return [...basic, ...basic, // two copies of the basic stuff (total 96)
+		...[...[...Array(4)].map((v, i) => {
+			return {colour: colours[i], value: "0", name: colours[i] + " 0"};
+		})], // the 4 0s
+		...[...[...Array(4)].map(v => {
+			return {colour: "Black", value: "Wild", name: "Wild"};
+		})], // wild cards
+		...[...[...Array(4)].map(v => {
+			return {colour: "Black", value: "+4", name: "Wild +4"};
+		})], // wild +4 cards
+	]; // 108 cards
+}
+
+// Durstenfeld shuffle algorithm - optimized Fisher-Yates shuffle.
+function shuffle(array) {
+	for (let i = array.length - 1; i > 0; i--) {
+		let j = Math.floor(Math.random() * (i + 1));
+		let temp = array[i];
+		array[i] = array[j];
+		array[j] = temp;
+	}
+	return array;
+}
+
+class UNOgame extends Rooms.RoomGame {
+	constructor(room, cap) {
+		super(room);
+
+		if (room.gameNumber) {
+			room.gameNumber++;
+		} else {
+			room.gameNumber = 1;
+		}
+
+		this.allowRenames = true;
+		this.playerCap = parseInt(cap) || 6;
+		this.maxTime = maxTime;
+
+		this.gameid = "uno";
+		this.title = "UNO";
+
+		this.state = "signups";
+
+		this.currentPlayer = null;
+		this.deck = shuffle(createDeck());
+		this.discards = [];
+		this.topCard = null;
+
+		this.direction = 1;
+	}
+
+	onStart() {
+		if (this.playerCount < 2) return false;
+		this.sendToRoom(`|uhtmlchange|uno-${this.room.gameNumber}|<div class="infobox">The game of UNO has started.</div>`);
+		this.state = "play";
+
+		this.onNextPlayer();  // determines the first player
+
+		// give cards to the players
+		for (let i in this.players) {
+			this.players[i].hand.push(...this.drawCard(7));
+		}
+
+		// top card of the deck.
+		do {
+			this.topCard = this.drawCard(1)[0];
+			this.discards.unshift(this.topCard);
+		} while (this.topCard.colour === "Black");
+
+		this.sendToRoom(`|raw|The top card is <font color=${this.topCard.colour}>${this.topCard.name}</font>.`);
```

Comment:
I assume that the alternative is <span>?

on this note, i'll remove all <center> as well and replace it with text-align iirc.

sorry didnt realize these weren't HTML5 supported.